src/cdn_definitions/_impl/__init__.py

```diff
@@ -18,19 +18,20 @@ def parsed(data, ext):
     )
 
 
-def get_remote_data(url):
+def get_remote_data(url, session=None):
     """
     Creates a requests session with retries. If the request was successful, returns the response.
     """
     retry_strategy = Retry(
         status_forcelist=[429, 500, 502, 503, 504],
     )
-    adapter = requests.adapters.HTTPAdapter(max_retries=retry_strategy)
-    session = requests.Session()
-    session.mount(url, adapter)
 
-    response = session.get(url, timeout=10)
+    if not isinstance(session, requests.Session):
```

Comment:
Hmm, I think this should really be "if not session".  Checking the concrete type before using an object is generally considered an antipattern I think, with duck typing preferred, though sadly I couldn't find any hints about this in official docs.

Basically, if someone passed you a session then you should just try to use it - if the object is compatible with the requests.Session interface, then it'll work.  If it isn't then it'll crash and that will be the caller's fault.

Consider for example that if someone passes you an instance of `Mock` where they've mocked out the returned value of get, then this code would silently ignore the passed object instead of using it since it's not precisely a subclass of requests.Session.